components/AppBar/src/MDCAppBarNavigationController.h

```diff
@@ -0,0 +1,83 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#ifndef MDC_SUBCLASSING_RESTRICTED
+#if defined(__has_attribute) && __has_attribute(objc_subclassing_restricted)
+#define MDC_SUBCLASSING_RESTRICTED __attribute__((objc_subclassing_restricted))
+#else
+#define MDC_SUBCLASSING_RESTRICTED
+#endif
+#endif  // #ifndef MDC_SUBCLASSING_RESTRICTED
+
+@class MDCAppBar;
+@class MDCAppBarNavigationController;
+
+/**
+ Defines the events that an MDCAppBarNavigationController may send to a delegate.
+ */
+@protocol MDCAppBarNavigationControllerDelegate <UINavigationControllerDelegate>
+@optional
```

Comment:
Why make this optional? Since there's only a single method, shouldn't delegates be warned when they fail to implement the protocol methods?

Reply:
Because this is an extension of UINavigationControllerDelegate and we don't want to break any existing implementations of the navigation controller delegate.

Reply:
Got it, I failed to notice the superprotocol.